### PR TITLE
Changed outdated type-function to is_<type> and stdlib version to 4.2

### PR DESCRIPTION
**Pull request by ejabach**

### CODE REVIEW

File: .fixtures.yml [hunk 1]
```diff
@@ -2,6 +2,6 @@ fixtures:
   repositories:
     'stdlib':
       repo: 'git://github.com/puppetlabs/puppetlabs-stdlib.git'
-      ref: '3.2.0'
+      ref: '4.2.0'
```
Comment (ghoneycutt) [inline .fixtures.yml]: Changing this dependency is not strictly needed here as the functions you are using are present in stdlib 3.2.0. Because of this, it should be in a separate commit, since it is a separate logical change. If you have to put `and` in your commit message, it is a good sign you should have multiple commits.

File: manifests/init.pp [hunk 2]
```diff
@@ -10,18 +10,12 @@
...
-    'string': {
-      $enable_hiera_array_real = str2bool($enable_hiera_array)
-    }
-    'boolean': {
-      $enable_hiera_array_real = $enable_hiera_array
-    }
-    default: {
-      fail("utils::enable_hiera_array must be of type boolean or string. Detected type is <${enable_hiera_array_type}>.")
-    }
+    if is_string($enable_hiera_array) {
```
Comment (ghoneycutt) [inline manifests/init.pp]: style issue: the line is two space too far to the right

File: Modulefile [hunk 3]
```diff
@@ -7,4 +7,4 @@ summary 'Manage utils'
 description 'Ensure arbitrary list of utility packages is present'
 project_page 'https://github.com/ghoneycutt/puppet-module-utils'
 
-dependency 'puppetlabs/stdlib', '>= 3.2.0'
+dependency 'puppetlabs/stdlib', '>=4.2.0'
```
Comment (ghoneycutt) [inline Modulefile]: This should actually be the following so that it is bounded. If not, the Puppet Forge docks you points on the quality scale.

``` ruby
dependency 'puppetlabs/stdlib', '>= 4.2.0 < 5.0.0'
```

File: .fixtures.yml (hunk 1 above)
Comment (ghoneycutt) [inline .fixtures.yml]: We should keep this at 3.2.0 though unless there is a real need to change.

Comment (ejabach) [inline .fixtures.yml]: Actually it is used in manifest/init.pp:15. is_bool() needs the newer stdlib version.

**Comment by ejabach** [comment]
Since the newer stdlib version is used I hope it is ok to have the changes in the same commit. If not, just let me know.

File: manifests/init.pp (hunk 2 above)
Comment (ejabach) [inline manifests/init.pp]: Done

**Comment by ejabach** [comment]
I'm having these failed checks with several modules, but I am not sure what it means. It crashes with Ruby version 1.8.7.

File: .fixtures.yml (hunk 1 above)
Comment (ghoneycutt) [inline .fixtures.yml]: Aha! I didn't notice that. the newer stdlib is fine in this PR.

**Comment by ghoneycutt** [comment]
rspec updated, so we need these lines in the Gemfile

https://github.com/ghoneycutt/puppet-module-hosts/blob/master/Gemfile#L15-18

**Comment by ejabach** [comment]
Okay, thanks. I'll update it where it's necessary.

**Comment by Phil-Friderici** [comment]
thanks @ejabach, looks good

**Comment by Phil-Friderici** [comment]
@ejabach Puppet v4 compatibility was reached with this commit: https://github.com/ghoneycutt/puppet-module-utils/commit/4aa572b0c064c692d65ceb45ffde20ba27c6b145

Obviosly, I recommend to close this PR now. But again, thanks for your work !

**Comment by ghoneycutt** [comment]
Thanks guys!
